.travis.yml

```diff
@@ -59,6 +62,7 @@ jobs:
           dist: xenial
           before_install:
               - CC=clang-7 CXX=clang++-7
+              - CXXFLAGS="$CXXFLAGS -Wno-sign-compare -Wno-narrowing"
```

Comment:
I like applying suppressions like these only where they are really needed (i.e., here in this file, for this compiler). :+1: 

But it looks to me that GCC also complains about narrowing, but the build doesn't fail because `-Werror` wasn't carried over from the Makefile?

Reply:
But we also need these flags for gcc on xenial, right? At least this is why the build fails currently.